boom/legacy.py

```
@@ -171,25 +183,35 @@ def write_legacy_loader(selection=None, loader=BOOM_LOADER_GRUB1, cfg_path=None)
                 dbe = decorator(be)
                 tmp_f.write(str(dbe) + "\n")
             tmp_f.write(end_tag + "\n")
-    except BoomLegacyFormatError as e:
-        _log_error("Error formatting %s configuration: %s", name, e)
+    except BoomLegacyFormatError as err:
+        _log_error("Error formatting %s configuration: %s", name, err)
         try:
-            unlink(tmp_path)
+            close(fd)
```

Ruff (0.12.2)

Undefined name `fd`

(F821)